### ISSUE TITLE
Error: implicit instantiation of undefined template during build

### ISSUE THREAD

**Issue by harrison-n** [fireice-uk/xmr-stak#365]
```
Scanning dependencies of target xmr-stak-c
[  3%] Building C object CMakeFiles/xmr-stak-c.dir/backend/cpu/crypto/c_blake256.c.o
[  6%] Building C object CMakeFiles/xmr-stak-c.dir/backend/cpu/crypto/c_groestl.c.o
[ 10%] Building C object CMakeFiles/xmr-stak-c.dir/backend/cpu/crypto/c_jh.c.o
[ 13%] Building C object CMakeFiles/xmr-stak-c.dir/backend/cpu/crypto/c_keccak.c.o
[ 17%] Building C object CMakeFiles/xmr-stak-c.dir/backend/cpu/crypto/c_skein.c.o
[ 20%] Linking C static library bin/libxmr-stak-c.a
[ 20%] Built target xmr-stak-c
Scanning dependencies of target xmr-stak-backend
[ 24%] Building CXX object CMakeFiles/xmr-stak-backend.dir/jconf.cpp.o
/usr/local/bin/xmr-stak/xmrstak/jconf.cpp:161:1: warning: control may reach end of non-void function [-Wreturn-type]
}
^
1 warning generated.
[ 27%] Building CXX object CMakeFiles/xmr-stak-backend.dir/version.cpp.o
[ 31%] Building CXX object CMakeFiles/xmr-stak-backend.dir/backend/cpu/hwlocMemory.cpp.o
[ 34%] Building CXX object CMakeFiles/xmr-stak-backend.dir/backend/cpu/jconf.cpp.o
[ 37%] Building CXX object CMakeFiles/xmr-stak-backend.dir/backend/cpu/minethd.cpp.o
[ 41%] Building CXX object CMakeFiles/xmr-stak-backend.dir/backend/backendConnector.cpp.o
[ 44%] Building CXX object CMakeFiles/xmr-stak-backend.dir/backend/globalStates.cpp.o
[ 48%] Building CXX object CMakeFiles/xmr-stak-backend.dir/backend/cpu/crypto/cryptonight_common.cpp.o
[ 51%] Building CXX object CMakeFiles/xmr-stak-backend.dir/http/httpd.cpp.o
[ 55%] Building CXX object CMakeFiles/xmr-stak-backend.dir/http/webdesign.cpp.o
[ 58%] Building CXX object CMakeFiles/xmr-stak-backend.dir/misc/console.cpp.o
[ 62%] Building CXX object CMakeFiles/xmr-stak-backend.dir/misc/executor.cpp.o
[ 65%] Building CXX object CMakeFiles/xmr-stak-backend.dir/misc/telemetry.cpp.o
[ 68%] Building CXX object CMakeFiles/xmr-stak-backend.dir/misc/utility.cpp.o
[ 72%] Building CXX object CMakeFiles/xmr-stak-backend.dir/net/jpsock.cpp.o
[ 75%] Building CXX object CMakeFiles/xmr-stak-backend.dir/net/socket.cpp.o
[ 79%] Linking CXX static library bin/libxmr-stak-backend.a
[ 79%] Built target xmr-stak-backend
Scanning dependencies of target xmr-stak
[ 82%] Building CXX object CMakeFiles/xmr-stak.dir/cli/cli-miner.cpp.o
[ 86%] Linking CXX executable bin/xmr-stak
[ 86%] Built target xmr-stak
Scanning dependencies of target xmrstak_opencl_backend
[ 89%] Building CXX object CMakeFiles/xmrstak_opencl_backend.dir/backend/amd/amd_gpu/gpu.cpp.o
[ 93%] Building CXX object CMakeFiles/xmrstak_opencl_backend.dir/backend/amd/jconf.cpp.o
[ 96%] Building CXX object CMakeFiles/xmrstak_opencl_backend.dir/backend/amd/minethd.cpp.o
In file included from /usr/local/bin/xmr-stak/xmrstak/backend/amd/minethd.cpp:24:
In file included from /usr/local/bin/xmr-stak/xmrstak/backend/amd/minethd.hpp:3:
/usr/local/bin/xmr-stak/xmrstak/backend/amd/amd_gpu/gpu.hpp:37:14: error: implicit instantiation of undefined template 'std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> >'
        std::string name;
                    ^
/Library/Developer/CommandLineTools/usr/include/c++/v1/iosfwd:193:32: note: template is declared here
    class _LIBCPP_TEMPLATE_VIS basic_string;
                               ^
1 error generated.
make[2]: *** [CMakeFiles/xmrstak_opencl_backend.dir/backend/amd/minethd.cpp.o] Error 1
make[1]: *** [CMakeFiles/xmrstak_opencl_backend.dir/all] Error 2
make: *** [all] Error 2
```

Appreciate everything you guys are doing. Trying to get this running on my hackintosh.

**Comment by drewwiley** [comment]
This should help you out: https://github.com/fireice-uk/xmr-stak/issues/158, just add the line `#include <string>` between  `#include <stdint.h>` and `#include <vector>` in /usr/local/bin/xmr-stak/xmrstak/backend/amd/amd_gpu/gpu.hpp

**Comment by fireice-uk** [comment]
This should be resolved in 2.1
